front/pages/w/[wId]/builder/assistants/index.tsx

```diff
@@ -153,208 +174,196 @@ export default function WorkspaceAssistants({
+  owner,
+  agents,
+  tabScope,
+  setShowDetails,
+  handleToggleAgentStatus,
+  showDisabledFreeWorkspacePopup,
+  setShowDisabledFreeWorkspacePopup,
+}: {
+  owner: WorkspaceType;
+  agents: LightAgentConfigurationType[];
+  tabScope: AgentConfigurationScope;
+  setShowDetails: (agent: LightAgentConfigurationType) => void;
+  handleToggleAgentStatus: (
+    agent: LightAgentConfigurationType
+  ) => Promise<void>;
+  showDisabledFreeWorkspacePopup: string | null;
+  setShowDisabledFreeWorkspacePopup: (s: string | null) => void;
+}) {
+  const router = useRouter();
+  if (tabScope === "published") {
+    return (
+      <div className="grid grid-cols-1 gap-2 sm:grid-cols-2">
+        {agents.map((a) => (
+          <AssistantPreview2
+            key={a.sId}
+            title={a.name}
+            pictureUrl={a.pictureUrl}
+            subtitle={a.lastAuthors?.join(", ") ?? ""}
+            description={a.description}
+            variant="list"
+            onClick={() => setShowDetails(a)}
+          />
+        ))}
+      </div>
+    );
+  }
+
+  return (
+    <ContextItem.List>
+      {agents.map((agent) => (
+        <ContextItem
+          key={agent.sId}
+          title={`@${agent.name}`}
+          subElement={
+            agent.scope === "global"
+              ? null
+              : assistantUsageMessage({
+                  assistantName: agent.name,
+                  usage: agent.usage || null,
+                  isLoading: false,
+                  isError: false,
+                  shortVersion: true,
+                })
+          }
+          visual={<Avatar visual={<img src={agent.pictureUrl} />} size="md" />}
+          onClick={() => setShowDetails(agent)}
+          action={
+            agent.scope === "global" ? (
+              <GlobalAgentAction agent={agent} />
+            ) : null
+          }
+        >
+          <ContextItem.Description>
+            <div className="text-element-700">{agent.description}</div>
+          </ContextItem.Description>
+        </ContextItem>
+      ))}
+    </ContextItem.List>
+  );
+
+  function GlobalAgentAction({
+    agent,
+  }: {
+    agent: LightAgentConfigurationType;
+  }) {
+    if (agent.sId === "helper") {
+      return null;
+    }
+
+    if (agent.sId === "dust") {
+      return (
+        <Button
+          variant="secondary"
+          icon={Cog6ToothIcon}
+          label="Manage"
+          size="sm"
+          disabled={!isBuilder(owner)}
+          onClick={(e) => {
+            e.stopPropagation();
+            void router.push(`/w/${owner.sId}/builder/assistants/dust`);
+          }}
+        />
+      );
+    }
+
+    return (
+      <div className="relative">
+        <SliderToggle
+          size="xs"
+          onClick={async (e) => {
+            e.stopPropagation();
+            await handleToggleAgentStatus(agent);
+          }}
+          selected={agent.status === "active"}
+          disabled={agent.status === "disabled_missing_datasource"}
```

Comment:
Shouldn't we gate this behind the `isBuilder` like before?

Reply:
Good point!